modules/node-shared/src/main/scala/io/constellationnetwork/node/shared/app/TessellationIOApp.scala

```diff
@@ -55,7 +54,8 @@ abstract class TessellationIOApp[A <: CliMethod](
   header: String,
   clusterId: ClusterId,
   helpFlag: Boolean = true,
-  version: TessellationVersion = TessellationVersion.unsafeFrom("0.0.1")
+  version: TessellationVersion = TessellationVersion.unsafeFrom("0.0.1"),
+  createConfig: () => IO[ConfigObjectSource]
```

Comment:
I think you can replace it with `createConfig: => IO[ConfigObjectSource]` and (lazily) get the result. Doesn't matter much here because you call it only once.